doped/utils/qehacks.py

```
@@ -0,0 +1,56 @@
+# hacks.py
+import numpy as np
+from pymatgen.electronic_structure.core import Spin
+
+class PymatgenEspressoHacks:
+    """
+    Hacks for pymatgen.io.qe.PWxml to enable setting read-only properties
+    like `projected_eigenvalues` and `projected_magnetisation`.
+    """
+
+    @staticmethod
+    def make_property_settable(cls, prop_name, backing_attr=None):
```

Ruff (0.12.2)

First argument of a static method should not be named `cls`

(PLW0211)